stock-managment/src/main/java/org/dhis2/android/rtsm/ui/home/HomeActivity.java

```diff
@@ -128,15 +147,32 @@ private void setupComponents() {
 
         viewModel.getFacility().observe(this, ou -> facilityTextView.setText(ou.displayName()));
 
+
         facilityTextView.setOnItemClickListener((adapterView, view, position, rowId) ->
-                viewModel.setFacility((OrganisationUnit) facilityTextView.getAdapter().getItem(position))
-        );
+                {
+                    viewModel.setFacility((OrganisationUnit) facilityTextView.getAdapter().getItem(position));
+
+                    OrganisationUnit organisationUnit = (OrganisationUnit) facilityTextView.getAdapter().getItem(position);
 
+                    from = organisationUnit.displayName();
+
+                    setSubtitle(from, to);
```

Comment:
I think, that composing the subtitle can be done from ViewModel as it is presentation logic, in that way, we can reduce code resposibilities in the home activity.